File: App/General/Category/App/UIButton+App.m

#import "UIButton+App.h"

@implementation UIButton (App)

- (nullable NSString *)text {
    return self.currentTitle;
}

- (void)setText:(nullable NSString *)text {
    self.titleLabel.text = text;
    [self setTitle:text forState:UIControlStateNormal];
}

@end

File: App/General/Category/App/UIButton+App.swift
/**
 应用级别的便捷方法
 */
extension UIButton {
    
    // @MBDependency:3
    /**
     便捷读取/设置当前状态的文字
     
     为了和 UILabel 等控件的设置方式一致
     */
    @objc var text: String? {
        get {
            return currentTitle
        }
        set {
            setTitle(text, for: state)
        }
    }
}
